src/hotspot/os/aix/os_aix.cpp

```diff
@@ -3033,29 +3049,134 @@ void os::jfr_report_memory_info() {}
 
 // Simulate the library search algorithm of dlopen() (in os::dll_load)
 int os::Aix::stat64x_via_LIBPATH(const char* path, struct stat64x* stat) {
-  if (path[0] == '/' ||
-      (path[0] == '.' && (path[1] == '/' ||
-                          (path[1] == '.' && path[2] == '/')))) {
-    return stat64x(path, stat);
+  if (path == nullptr)
+    return -1;
+
+  char *path2 = strdup (path);
+  // if exist, strip off trailing (shr_64.o) or similar
+  int idx = strlen(path2) - 1;
+  if (path2[idx] == ')') {
+    while (path2[idx] != '(' && idx > 0) idx--;
+    if (idx > 0)
+      path2[idx] = 0;
+  }
+
+  int ret = -1;
+  if (path2[0] == '/' ||
+      (path2[0] == '.' && (path2[1] == '/' ||
+                          (path2[1] == '.' && path2[2] == '/')))) {
```

Comment:
This complexity is not needed, nor is it sufficient.

https://www.ibm.com/docs/en/aix/7.1?topic=d-dlopen-subroutine

"If FilePath contains a slash character, FilePath is used directly, and no directories are searched. "

So, just scan for a '/' - if you find one, its a path to be opened directly:

```
const bool use_as_filepath = strchr(path2, '/'); 
```